### PR TITLE
Feature/bbb 660 refactor user ids

### PR DESCRIPTION
**Pull request by pauljroberts**

### CODE REVIEW

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementApiClient.java [hunk 1]
```diff
@@ -45,25 +46,28 @@ public UserManagementApiClient(
...
     ResponseEntity<UsersResponse> userListResponse =
         restTemplate.getForEntity(
-            GET_USERS_FOR_AUTHORITY_ENDPOINT, UsersResponse.class, nameFilter, authorityId);
+            GET_USERS_FOR_AUTHORITY_ENDPOINT, UsersResponse.class, nameFilter, authorityShortCode);
     return Objects.requireNonNull(userListResponse.getBody()).getData();
   }
 
-  public User getById(Integer userId) {
-    Assert.notNull(userId, "userId must be provided for getById");
+  public User getByUuid(UUID userId) {
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementApiClient.java]: Could have renamed variable, but I don't feel strongly about it.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/CreateANewUserFormRequestToUser.java [hunk 2]
```diff
@@ -10,8 +10,11 @@
 public class CreateANewUserFormRequestToUser implements Converter<CreateANewUserFormRequest, User> {
 
   @Override
-  public User convert(CreateANewUserFormRequest source) {
-    Assert.notNull(source, "Source cannot be null");
-    return new User().name(source.getName()).emailAddress(source.getEmailAddress());
+  public User convert(CreateANewUserFormRequest formRequest) {
+    Assert.notNull(formRequest, "Source cannot be null");
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/CreateANewUserFormRequestToUser.java]: Message says source

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/UserDetailsFormRequestToUser.java [hunk 3]
```diff
@@ -10,8 +10,11 @@
 public class UserDetailsFormRequestToUser implements Converter<UserDetailsFormRequest, User> {
 
   @Override
-  public User convert(UserDetailsFormRequest source) {
-    Assert.notNull(source, "Source cannot be null");
-    return new User().name(source.getName()).emailAddress(source.getEmailAddress());
+  public User convert(UserDetailsFormRequest formRequest) {
+    Assert.notNull(formRequest, "Source cannot be null");
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/UserDetailsFormRequestToUser.java]: Message contains 'source'

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/request/UserDetailsFormRequest.java [hunk 4]
```diff
@@ -4,7 +4,8 @@
 
 @Data
 public class UserDetailsFormRequest {
-  private Integer localAuthorityId;
+  //  private Integer localAuthorityId;
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/request/UserDetailsFormRequest.java]: Commented out code not needed.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/service/UserService.java [hunk 5]
```diff
@@ -47,15 +48,15 @@ public User update(User user) {
     return userManagementApiClient.updateUser(user);
   }
 
-  public User updatePassword(String uuid, String password, String passwordConfirm) {
+  public User updatePassword(UUID uuid, String password, String passwordConfirm) {
     return setPasswordApiClient.updatePassword(uuid, password, passwordConfirm);
   }
 
-  public void delete(Integer id) {
+  public void delete(UUID id) {
```
Comment (pauljroberts) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/service/UserService.java]: I suppose id works, but uuid or userUuid might be better.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementApiClient.java (hunk 1 above)
Comment (robsvaltech) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementApiClient.java]: done

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/CreateANewUserFormRequestToUser.java (hunk 2 above)
Comment (robsvaltech) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/CreateANewUserFormRequestToUser.java]: done

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/UserDetailsFormRequestToUser.java (hunk 3 above)
Comment (robsvaltech) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/converter/requesttoservice/UserDetailsFormRequestToUser.java]: done

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/request/UserDetailsFormRequest.java (hunk 4 above)
Comment (robsvaltech) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/controller/request/UserDetailsFormRequest.java]: done

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/service/UserService.java (hunk 5 above)
Comment (robsvaltech) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/service/UserService.java]: done

**Pull request merged**